### ISSUE TITLE
Generate types inside a theme package

### ISSUE THREAD

**Issue by BryceRussell** [astrolicious/astro-theme-provider#120]
When developing a theme, types are generated inside a playground. This makes it hard to properly link types while developing a theme. To improve this, types could also be generated inside the theme package to prevent issues with hoisting or circular type references.

**Things to consider**:
- Types are generated using the [AIK utility `addDts`](https://astro-integration-kit.netlify.app/utilities/add-dts/). This utility only generates types inside a user project, this issue would require a custom utility (or an upstream addition to AIK) for type generation inside a package
- Where are these types generated inside a package? `virtual.d.ts`?

**Comment by BryceRussell** [comment]
I am going to take over this issue and rename it, I worded it poorly and didn't explain exactly why I created it. Generating types inside the theme's package may have some advantages, but I don't know of any right now.

The real issue here is that theme authors need Content Collections types while developing inside a package. 

This is a bit complicated because we need to generate the Content Collection types inside a playground and hoist them somehow so that authors have type safety when creating theme packages.

TLDR: Typically, types flow downwards from packages to users. We need to pass types upwards from a user to a package.
